### PR TITLE
add handling for linux

### PR DESCRIPTION
**Pull request by EIA485**

### CODE REVIEW

File: NeosModLoader/Configuration.cs [hunk 1]
```diff
@@ -15,7 +15,12 @@ internal static Configuration get()
             if (_configuration == null)
             {
                 // the config file can just sit next to the dll. Simple.
-                string path = $"{GetAssemblyDirectory()}\\{CONFIG_FILENAME}";
+                string path;
+                if (Environment.OSVersion.Platform==PlatformID.Unix)
+                    path = $"{GetAssemblyDirectory()}/{CONFIG_FILENAME}";
+                else
+                    path = $"{GetAssemblyDirectory()}\\{CONFIG_FILENAME}";
+
```
Comment (zkxs) [inline NeosModLoader/Configuration.cs:23]: If we're going to be handling cross-platform paths, we might as well do it the right way by using [Path.Combine](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=net-5.0) as discussed in this [stackoverflow post](https://stackoverflow.com/questions/5758633/how-should-i-handle-windows-linux-paths-in-c-sharp)

File: NeosModLoader/NeosModLoader.csproj [hunk 2]
```diff
@@ -42,7 +42,6 @@
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
-    <Reference Include="System.Data.DataSetExtensions" />
```
Comment (zkxs) [inline NeosModLoader/NeosModLoader.csproj:45]: What's up with this?

File: NeosModLoader/ModLoader.cs [hunk 3]
```diff
@@ -22,7 +22,12 @@ internal static void LoadMods()
                 return;
             }
 
-            string modDirectory = Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName + "\\nml_mods";
+            string modDirectory;
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
+                modDirectory = System.IO.Path.GetDirectoryName(Assembly.GetEntryAssembly().Location)+"/nml_mods";
+            else
+                modDirectory = Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName + "\\nml_mods";
+
```
Comment (zkxs) [inline NeosModLoader/ModLoader.cs:30]: If we're going to be handling cross-platform paths, we might as well do it the right way by using [Path.Combine](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=net-5.0) as discussed in this [stackoverflow post](https://stackoverflow.com/questions/5758633/how-should-i-handle-windows-linux-paths-in-c-sharp)

**Review by zkxs — CHANGES_REQUESTED** [review]

File: NeosModLoader/NeosModLoader.csproj (hunk 2 above)
Comment (EIA485) [reply NeosModLoader/NeosModLoader.csproj:45]: i accidentally removed the reference and it still compiled so i did not think much of it, if you would like im happy to add that reference back.

Comment (zkxs) [reply NeosModLoader/NeosModLoader.csproj:45]: I'm curious what the reason for removing it was? Is it incompatible with Linux somehow?

File: NeosModLoader/ModLoader.cs (hunk 3 above)
Comment (EIA485) [reply NeosModLoader/ModLoader.cs:30]: unless you know a better way, i need to handle getting neos's path differently on windows than on Linux

Comment (zkxs) [reply NeosModLoader/ModLoader.cs:30]: Is there a reason that this won't work for both windows and linux?
```cs
string modDirectory = Paths.Combine(Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName, "nml_mods");
```

Comment (EIA485) [reply NeosModLoader/ModLoader.cs:30]: not sure why but that fails on linux, it seems Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName
is null on linux

Comment (zkxs) [reply NeosModLoader/ModLoader.cs:30]: Weird... can you figure out where it's returning null? None of the docs for anything in that chain indicate nulls can be returned.

Alternatively, if `Path.GetDirectoryName(Assembly.GetEntryAssembly().Location)` works on Windows as well lets just ditch the Process API completely.

**Review by zkxs — APPROVED** [review]

**Pull request merged**